Applications/Spire/Source/Ui/Button.cpp

```diff
@@ -0,0 +1,138 @@
+#include "Spire/Ui/Button.hpp"
+#include <QHBoxLayout>
+#include <QMouseEvent>
+#include <QEvent>
+#include "Spire/Ui/TextBox.hpp"
+#include "Spire/Spire/Dimensions.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+Button::Button(QWidget* component, QWidget* parent)
+    : QWidget(parent),
+      m_component(component),
+      m_is_down(false) {
+  auto layout = new QHBoxLayout(this);
+  layout->setContentsMargins(0, 0, 0, 0);
+  layout->addWidget(m_component);
+  m_component->installEventFilter(this);
+  setFocusPolicy(Qt::StrongFocus);
+  setObjectName("Button");
+  setStyleSheet(QString(R"(
+    #Button {
+      background-color: #FFFFFF;
+      border: %1px solid #C8C8C8;
+    }
+    #Button:hover {
+      border-color: #4B23A0;
+    }
+    #Button:focus {
+      border-color: #4B23A0;
+    }
+    #Button:disabled {
+      background-color: #F5F5F5;
+      border-color: #C8C8C8;
+    })").arg(scale_width(1)));
+}
+
+connection Button::connect_clicked_signal(
+    const ClickedSignal::slot_type& slot) const {
+  return m_clicked_signal.connect(slot);
+}
+
+bool Button::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == m_component) {
+    switch(event->type()) {
+    case QEvent::MouseButtonPress:
```

Comment:
Give the component a chance to handle these events and only if the component doesn't handle them, then the Button can handle it.